include/aws/http/request_response.h

```diff
@@ -233,6 +233,12 @@ struct aws_http_make_request_options {
      * See `aws_http_on_stream_complete_fn`.
      */
     aws_http_on_stream_complete_fn *on_complete;
+
+    /**
+     * When using h2, request body data will be provided over time. The body stream on the request will only be
+     * read when body data has been supplied via `aws_http2_stream_write_data`
+     */
+    bool h2_use_evented_body_stream;
```

Comment:
We have different `aws_http2_connection_options` and `aws_http1_connection_options` for connection, not sure if we want the similar stuff for request. For now, I don't think we will need it as I don't see any more request related options specific to version. But, just bring it up to table 😀

Reply:
Well, these wouldn't be connection options for H2, these are more stream/request related.

Reply:
Oh, I was thinking about to have something like `aws_http1_make_request_options` and `aws_http2_make_request_options`, in case we will have more version specific options added. For now, I don't think we will need it as I don't see there will be much to add.

Reply:
That makes sense, we could do that at any point.